src/sync.ts

```diff
@@ -1792,3 +1720,49 @@ function createNewUser(client: MatrixClient, userId: string): User {
     return user;
 }
 
+// /!\ This function is not intended for public use! It's only exported from
+// here in order to share some common logic with sliding-sync-sdk.ts.
+export function createRoom(client: MatrixClient, roomId: string, opts: Partial<IStoredClientOpts>): Room {
```

Comment:
I wonder if we could name this a bit better to avoid someone using it by accident - `createRoomAndSetupClientReEmitter()`?